src/beamlime/logging/records.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from logging import LogRecord
+from types import TracebackType
+from typing import Mapping, Union, overload
+
+from colorama import Style
+
+from .formatters import EXTERNAL_MESSAGE_HEADERS
+
+_SysExcInfoType = Union[
+    tuple[type[BaseException], BaseException, Union[TracebackType, None]],
+    tuple[None, None, None],
+]
+
+_ArgsType = Union[tuple[object, ...], Mapping[str, object]]
+
+
+class BeamlimeLogRecord(LogRecord):
+    """
+    Note
+    ----
+    First 8 Arguments cannot be keyword-only arguments
+    because of the way ``logging`` creates an empty record.
+    It uses positional arguments to create a record instance.
+    """
```

Comment:
Not really a docstring. Can you turn this into a comment instead?

Reply:
Actually it's not true anymore. I will delete it.